aptos-move/framework/aptos-framework/sources/transaction_context.move

```diff
@@ -182,6 +189,15 @@ module aptos_framework::transaction_context {
         payload.entry_function_payload
     }
 
+    /// Returns the hash of the current raw transaction, which is used for transaction authentication.
+    /// This function calls an internal native function to retrieve the raw transaction hash.
+    /// In the case of a fee payer transaction, the hash value is generated using 0x0 as the fee payer address.
+    public fun raw_transaction_hash(): vector<u8> {
+        assert!(features::transaction_context_hash_function_update_enabled(), error::invalid_state(ETRANSACTION_CONTEXT_HASH_FUNCTION_UPDATE_NOT_ENABLED));
+        raw_transaction_hash_internal()
```

Comment:
I'm actually not super familiar with why we do this. In general, why do we separate the native function into two functions with `_internal` tag?

Reply:
It would've been ideal to just add `public native fun raw_transaction_hash()`. However, it's turns out to be unsafe during the release process due to the timing not being perfectly aligned between the binary upgrade and the framework upgrade.